draft-jmiller-json-proof-algorithms.md

```diff
@@ -184,15 +184,163 @@ The Verifier uses only the disclosed payloads and generates or uses the included
+```
+
+For verification a nonce is needed:
+```json
+[127, 207, 97, 120, 199, 184, 23, 159, 105, 123, 246, 10, 55, 7, 221, 96, 213, 104, 73, 98, 217, 32, 121, 237, 162, 90, 150, 9, 180, 72, 226, 188]
+```
+
+To generate a proof, the `blsCreateProof()` method is used with a revealed indexes array argument of `[0, 2, 4]` and results in the octet string:
+```json
+[0, 5, 21, 161, 191, 254, 45, 60, 183, 31, 8, 229, 8, 160, 190, 97, 252, 230, 185, 215, 158, 121, 125, 186, 81, 20, 232, 4, 201, 86, 118, 56, 241, 236, 207, 206, 210, 82, 31, 229, 175, 239, 6, 193, 68, 14, 80, 208, 205, 219, 106, 165, 188, 183, 147, 59, 186, 90, 209, 11, 27, 44, 167, 126, 166, 101, 132, 205, 207, 22, 35, 181, 148, 118, 44, 164, 163, 58, 16, 211, 245, 133, 115, 158, 158, 89, 182, 5, 151, 101, 91, 28, 59, 75, 224, 207, 48, 75, 220, 131, 79, 188, 57, 235, 164, 122, 170, 120, 251, 230, 158, 230, 73, 152, 138, 106, 61, 106, 90, 215, 124, 158, 179, 82, 2, 137, 107, 96, 130, 131, 215, 25, 41, 242, 146, 48, 158, 82, 61, 185, 135, 172, 63, 248, 149, 50, 183, 0, 0, 0, 116, 179, 185, 176, 214, 11, 160, 24, 235, 157, 189, 141, 33, 225, 119, 5, 146, 167, 175, 227, 188, 165, 196, 210, 156, 216, 164, 28, 167, 32, 97, 215, 36, 122, 137, 173, 182, 17, 233, 69, 41, 20, 44, 69, 203, 236, 226, 24, 242, 0, 0, 0, 2, 0, 199, 248, 173, 124, 33, 127, 253, 79, 170, 77, 153, 101, 169, 100, 88, 190, 206, 55, 24, 25, 80, 108, 215, 80, 134, 128, 70, 190, 240, 209, 148, 67, 202, 130, 180, 145, 51, 72, 188, 140, 185, 142, 193, 246, 155, 12, 248, 65, 14, 209, 72, 248, 119, 250, 123, 46, 136, 8, 191, 45, 173, 77, 125, 180, 76, 253, 10, 139, 244, 156, 138, 226, 149, 41, 76, 50, 45, 210, 231, 74, 18, 192, 91, 36, 154, 46, 158, 76, 243, 13, 119, 122, 107, 206, 91, 226, 160, 27, 80, 206, 97, 32, 74, 42, 155, 180, 137, 229, 210, 80, 236, 0, 0, 0, 4, 6, 143, 167, 94, 118, 29, 1, 252, 243, 190, 176, 42, 45, 175, 213, 182, 81, 7, 250, 244, 141, 136, 168, 27, 164, 110, 101, 28, 181, 237, 236, 54, 12, 67, 44, 104, 187, 33, 249, 159, 143, 104, 204, 234, 0, 251, 101, 79, 170, 206, 113, 44, 235, 62, 242, 254, 139, 252, 218, 73, 155, 48, 127, 131, 79, 41, 134, 186, 183, 174, 247, 112, 103, 33, 131, 82, 115, 233, 36, 235, 12, 179, 171, 71, 55, 154, 86, 221, 77, 206, 171, 190, 186, 31, 192, 43, 88, 0, 45, 109, 204, 217, 154, 132, 113, 93, 166, 177, 51, 31, 118, 153, 88, 116, 45, 113, 111, 187, 7, 150, 214, 184, 244, 198, 149, 222, 93, 101]
+```
+
+The resulting verifiable JWP in JSON serialization is:
+```json
+{
+  "protected": "eyJpc3MiOiJodHRwczovL2lzc3Vlci50bGQiLCJjbGFpbXMiOlsiZmFtaWx5X25hbWUiLCJnaXZlbl9uYW1lIiwiZW1haWwiLCJhZ2UiXSwidHlwIjoiSlBUIiwiYWxnIjoiQkJTLVgifQ",
+  "payloads": [
+    null,
```

Comment:
so you preserve revealed index locations with nulls correct?

Reply:
Yes, in the JSON serialization a [null](https://github.com/json-web-proofs/json-web-proofs/blob/main/draft-jmiller-json-web-proof.md?plain=1#L141) is used to represent non-disclosed payloads.